### PR TITLE
remove go-multierror lib

### LINKED ISSUE #63: remove go-multierror lib
snyk doesn't like the MPL 2.0 license that hashicorp/go-multierror uses

### PR DESCRIPTION
**Pull request by theganyo**
Fixes #63

### CODE REVIEW

File: errorset/errorset.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 )
 
 func Append(err error, errs ...error) error {
-	if err == nil && len(errs) == 1 && errs[0] == nil {
+	if err == nil && (len(errs) == 0 || len(errs) == 1 && errs[0] == nil) {
```
Comment (rockspore) [inline errorset/errorset.go:23]: I don't fully understand this and the corresponding test. So technically you could call Append(nil, nil, nil) and it should still return nil but that's not checked here.

Comment (rockspore) [reply errorset/errorset.go:23]: Could this be changed to call Append() recursively to make it more generic?

Comment (theganyo) [reply errorset/errorset.go:23]: Yeah, I know. I was being lazy. I'll do this proper.

Comment (rockspore) [reply errorset/errorset.go]: Thanks! ;-)

**Review by rockspore — APPROVED** [review]

**Pull request merged**